### ISSUE TITLE
Fix S1854 FP: Throw should visit finally

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#9466]
We need to add a branch for Throw semantics to connect the blocks
```csharp
    public int ReadInFinallyAfterCatch()
    {
        var value = 0;
        try
        {
            CanThrow();
            value = 42;
        }
        catch
        {
            value = 1;  // Noncompliant FP, used in finally after the throw
            throw;
        }
        finally
        {
            Log(value);
        }
        return value;
    }
```

**Comment by levanoz** [comment]
This false positive still exists in version 9.32

**Comment by mary-georgiou-sonarsource** [comment]
Hello @levanoz!
could you please open a new issue with the rule affected and a reproducer?
Thanks a lot.
